packages/app/src/elements/import-dialog.ts

```diff
@@ -67,13 +67,24 @@ export class ImportDialog extends Dialog<string, void> {
         `;
     }
 
-    async show(input: string) {
+    async show(file: File) {
         await this.updateComplete;
         const result = super.show();
-        this._rawData = input;
-        this._formatSelect.value = (imp.guessFormat(input) || imp.CSV).value;
-        this._parseString();
-        this._vaultSelect.value = app.mainVault!;
+
+        const reader = new FileReader();
```

Comment:
I experimented with, but ended up not liking passing the `FileReader` down the line. It meant we had to keep a reference of the file inside the `ImportDialog`, and there are no performance gains here, since we're not parsing 1pux/zip yet at this point, now.

Reply:
The problem was that the `FileReader` and reading of the file had to be implemented/called in many more places (for each type of import). It just seemed less clear/readable to me, but I'm happy to implement it that way, if you prefer.

Reply:
There are actually convenience functions for reading files already implemented in https://github.com/padloc/padloc/blob/master/packages/core/src/attachment.ts. You can move those to `@padloc/core/src/util` for reuse.

Reply:
And while you're at it maybe also give them all consistent names (i.e. `readFileAs...`).